src/enclave/provider.ts

```diff
@@ -23,23 +23,27 @@ export class EnclaveWSProvider implements EnclaveProvider {
 
 	private url: string;
 	private ws?: WebSocket;
+	private initialized: boolean;
 
 	constructor(url: URL) {
 		this.url = url.toString();
 		this.onopen = null;
 		this.onclose = null;
 		this.onerror = null;
 		this.onmessage = null;
+		this.initialized = false;
 	}
 
 	public connect() {
-		if (this.ws) {
-			this.ws.close();
-		}
+		this.close();
 		this.ws = new WebSocket(this.url);
 		this.ws.onmessage = this.onmessage;
 		this.ws.onerror = this.onerror;
-		this.ws.onopen = this.onopen;
+		this.ws.onopen = (e: Event): any => {
+			this.initialized = true;
+			if(this.onopen)
```

Comment:
`Prettier` kicked in for me here. Version `2.3.2`. Can you check this on your end please?

Reply:
The whole repository doesn't have prettier, yet. For me it currently lists over 70 files as ugly. I think we should make a PR that runs it over the whole repository and then also set up the compile-time check like we do in the marketplace.

Reply:
Ah, so I mixed that up with our demo then :laughing:

Reply:
Ironically we have a commit "incorporating prettier" but it didn't apply any changes to the files. We will do this in another PR, agree.